mongo/datadog_checks/mongo/mongo.py

```diff
@@ -25,6 +25,21 @@
 ALLOWED_CUSTOM_QUERIES_COMMANDS = ['aggregate', 'count', 'find']
 
 
+def build_url(scheme, host, path='/', username=None, password=None, query_params=None):
+    # type: (str, str, str, str, str, dict) -> str
+    """Build an URL from individual parts. Makes sure that parts are properly URL-encoded."""
+    if username and password:
```

Comment:
We should raise `ConfigurationError` is only `username` or `password` is set.

Reply:
> (not necessarily part of the fix though)

I agree it's a separate problem, I'll submit this as a follow up PR. We can probably include it in the RC because it's a small UX hole introduced by this new connection config parsing, but we'll see. :-)